tsl/src/continuous_aggs/materialize.c

```diff
@@ -398,22 +401,74 @@ build_merge_update_clause(List *column_names)
 	return ret->data;
 }
 
+static char *
+build_order_by_clause(MaterializationContext *context)
+{
+	if (!TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(context->mat_ht))
+		return ""; /* No ORDER BY if no compression */
+
+	CompressionSettings *settings = ts_compression_settings_get(context->mat_ht->main_table_relid);
+
+	int num_segmentby = ts_array_length(settings->fd.segmentby);
+	int num_orderby = ts_array_length(settings->fd.orderby);
+	int n_keys = num_segmentby + num_orderby;
```

Comment:
should probably have 2 dedicated loops here for segmentby and orderby since ASC/DESC and NULLS ordering only apply to the latter